File: tps/0/bench/haskell/app/Args.hs
module Args where
import System.Environment (getArgs)

getCLIArgs :: IO [Int]
getCLIArgs = do
  args <- getArgs
  -- we expect two value from the console, the first for the wmIterations and the second for the iterations
  let wmIterations | null args = 10000
                   | not (null (head args)) =  read (head args) :: Int
                   | otherwise = 10000
  let iterations | null args = 10000 
                 |  not (null (last args)) =  read (last args) :: Int
                 | otherwise = 10000
  return [wmIterations, iterations]

File: tps/0/bench/js/args.js
//argv[0] and argv[1] are the process execution path and the path for the js file respectively.
//So argv[2] is the first arg that you pass and so on...

// we expect two value from the console, the first for the warmup iterations and the second for the iterations
export const BENCH_ARGS = {
	WM_ITERATIONS: process.argv[2] || 10000,
	ITERATIONS: process.argv[3] || 10000,
};
